File: 프로그래머스/Level2/위장.py
# https://programmers.co.kr/learn/courses/30/lessons/42578

def solution(clothes):
    clothes_dict = dict()
    for name, kind in clothes:
        if kind in clothes_dict:
            clothes_dict[kind].append(name)
        else:
            clothes_dict[kind] = [name]
    answer = 1
    for value in clothes_dict.values():
        answer *= (len(value) + 1)
    answer -= 1
    return answer

File: 프로그래머스/Level2/위장.js
// https://programmers.co.kr/learn/courses/30/lessons/42578

function solution(clothes) {
  const hash = {};
  clothes.forEach((a) => {
    if (a[1] in hash) hash[a[1]]++;
    else hash[a[1]] = 1;
  });
  let answer = 1;
  for (const k in hash) {
    answer *= hash[k] + 1;
  }
  return answer - 1;
}
